test/extensions/filters/http/aws_request_signing/aws_request_signing_filter_test.cc

```diff
@@ -41,7 +43,7 @@ class AwsRequestSigningFilterTest : public testing::Test {
 // Verify filter functionality when signing works.
 TEST_F(AwsRequestSigningFilterTest, SignSucceeds) {
   setup();
-  EXPECT_CALL(*(filter_config_->signer_), sign(_));
+  EXPECT_CALL(*(filter_config_->signer_), sign(Matcher<Http::RequestHeaderMap&>(_), false));
```

Comment:
@rojkov I started working on the tests tonight only to realize that the new `sign` overload can't be disambiguated by the mocks.

I wanted to get feedback on this strategy before updating everywhere. I didn't see `Matcher` or the other recommended [techniques](https://google.github.io/googletest/gmock_cook_book.html#SelectOverload) used elsewhere in the envoy code.

Is this a good approach? If not, do you have a recommendation?

Reply:
This seems to be fine. `testing::Matcher` is actually used in some Envoy tests.

Reply:
Ah, thanks - I now see Matcher and others used throughout the tests. I must have had a stray search filter enabled when I was looking before.